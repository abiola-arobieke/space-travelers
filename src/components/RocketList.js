import { useSelector } from 'react-redux';
import RocketItem from './RocketItem';
import { addReservation, cancelReservation } from '../redux/rockets/rocketsSlice';

const RocketList = () => {
  const { rocketsItem } = useSelector((store) => store.rockets);

  return (
    <div>
      {rocketsItem.map((rocket) => (
        <RocketItem
          key={rocket.id}
          rocket={rocket}
          reservation={addReservation}
          cancelReservation={cancelReservation}
        />
      ))}
    </div>
  );
};

export default RocketList;
