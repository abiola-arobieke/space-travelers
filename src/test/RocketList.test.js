import React from 'react';
import renderer from 'react-test-renderer';
import { useSelector, useDispatch } from 'react-redux';
import RocketList from '../components/RocketList';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

jest.mock('../redux/rockets/rocketsSlice', () => ({
  addReservation: jest.fn(),
  cancelReservation: jest.fn(),
}));

describe('RocketList', () => {
  it('renders correctly', () => {
    const rocketsItem = [
      {
        id: 1,
        name: 'Falcon 9',
        description: 'Sample description 1',
        reserved: false,
        flickr_images: ['https://example.com/image1.jpg'],
      },
      {
        id: 2,
        name: 'Falcon Heavy',
        description: 'Sample description 2',
        reserved: true,
        flickr_images: ['https://example.com/image2.jpg'],
      },
    ];

    useSelector.mockReturnValue({ rocketsItem });
    const dispatchMock = jest.fn();
    useDispatch.mockReturnValue(dispatchMock);

    const component = renderer.create(<RocketList />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
