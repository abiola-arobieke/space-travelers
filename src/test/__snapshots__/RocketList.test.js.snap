// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RocketList renders correctly 1`] = `
<div>
  <div
    className="ds-flex card-container"
  >
    <div
      className="mid-screen-4"
    >
      <div
        className="card-image"
      >
        <img
          alt=""
          className="rocket-img"
          src="https://example.com/image1.jpg"
        />
      </div>
    </div>
    <div
      className="mid-screen-8 card-details"
    >
      <h2
        className="card-name"
      >
        Falcon 9
      </h2>
      <p
        className="card-description"
      >
        Sample description 1
      </p>
      <button
        className="btn-add"
        onClick={[Function]}
        type="button"
      >
        Add Reservation
      </button>
    </div>
  </div>
  <div
    className="ds-flex card-container"
  >
    <div
      className="mid-screen-4"
    >
      <div
        className="card-image"
      >
        <img
          alt=""
          className="rocket-img"
          src="https://example.com/image2.jpg"
        />
      </div>
    </div>
    <div
      className="mid-screen-8 card-details"
    >
      <h2
        className="card-name"
      >
        Falcon Heavy
      </h2>
      <p
        className="card-description"
      >
        <span
          className="pills-green"
        >
          reserved
        </span>
        Sample description 2
      </p>
      <button
        className="btn-cancel"
        onClick={[Function]}
        type="button"
      >
        Cancel Reservation
      </button>
    </div>
  </div>
</div>
`;
